File: pset1/greedy.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int total, counter;
void count(int coin)
{
    while (total>coin || total==coin)
{
    total=total-coin;
    counter++;
}
}
int main(void){
    
printf("O hai! ");
do{
printf("How much change is owed?\n");
    total=round(get_float()*100);
}
while (total<0);



count(25);
count(10);
count(5);
count(1);

printf("%i\n",counter);
}

File: pset6/greedy.py
import cs50


def main():
    print("O hai!How much change is owed?")
    while(True):
        total=round(cs50.get_float()*100)
        if total<0:
            print("Please input the amount of change in dollars")
        else:
            break

    print("Initial amount is {} cents".format(total))
    counter=0
    for i in [25, 10, 5, 1]:
        if total>=i:
            counter+=round(total/i)
            total=round(total%i)
        print("Current amount left is {} cents".format(total))
        print("Current number of coins is {}".format(counter))


if __name__ == "__main__":
    main()